src/facet/RewardsFacet.sol

```diff
@@ -79,6 +86,22 @@ contract RewardsFacet is Multicall, SafeOwnable {
         emit BadgeMinted(_token, _quantity, _to, _id, _activityType, _uri);
     }
 
+    function mintBadge(
+        address _token,
+        address _to,
+        uint256 _quantity,
+        bytes32 _id,
+        bytes32 _activityType,
+        string calldata _uri
```

Comment:
Is this the best names? I am not sure what each one of these does

can we rename to something like this? 
```solidity
address _badgeContract,
address _to,
uint256 _quantity,
bytes32 action,
bytes32 mission,
bytes calldata _data
```

`string calldata _uri` could become more versatile if `bytes calldata _data` is used but it really depends what the use of this is.

Reply:
I'm happy with your suggestions. "action" and "mission" are not quite contextually correct. The `_id` is an identifier for an `action` or `mission` (e.g complete activity), but it could be called `activityId`? The `_activityType` can either be `action` or `mission`. I called it `activityType` because I don't know if there will be more than just `actions` and `missions` in the future. Additionally, the `activityType` exists because the subgraph needs to determine whether it's an `action` or a `mission` ([Ref](https://linear.app/openformat/issue/OF-64/feature-view-badges#comment-0b1865ea)), and I couldn't think of another way to achieve this.

I hope this helps, happy to chat more about it on Monday.